Layouts/Base.inc

```diff
@@ -328,41 +328,48 @@ $endif
 ; ADC and temperature measurement
 ;**** **** **** **** ****
 ; Temperature measurement ADC value for which main motor power is limited at 80degC (low byte, assuming high byte is 1)
-IF MCU_48MHZ < 2
-	TEMP_LIMIT		    EQU	49
-ELSEIF MCU_48MHZ == 2
-	TEMP_LIMIT		    EQU	92	; 0degC is on the BB51 at 0x0114 so ((92 - 20) / 9 * 10) = 80degC
-ENDIF
+TEMP_LIMIT_2S		EQU	49  ; When using vdd 3.3V reference
+TEMP_LIMIT_1S       EQU 92  ; When using 1.65V internal reference
 TEMP_LIMIT_STEP		EQU	9		; Temperature measurement ADC value increment for another 10degC
 
 Initialize_Adc MACRO
-	IF MCU_48MHZ < 2
-		mov	REF0CN, #0Ch			;; Set vdd (3.3V) as reference. Enable temp sensor and bias
-		IF MCU_48MHZ == 0
+    LOCAL initialize_adc_use_internal_1V65_vref initialize_adc_use_vdd_3V3_vref initialize_adc_vref_selected
+    IF MCU_TYPE < 2
+	    mov Temp1, #Pgm_Power_Rating
```

Comment:
Indentation seems a bit off in this block.

Reply:
Reworked